### PR TITLE
[NOTIFICATIONS #1] feat: show order submitted notification from event

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Refactored submitted order notifications and bound them to `EVENT_EMITTER`.

1. In each trade flow (swapFlow,twapFlow,safeBundleEthFlow,etc) added call of `emitPostedOrderEvent()`
2. Created `OrdersNotificationsUpdater` which listens `ON_POSTED_ORDER` events and:
a. displays a snackbar
b. emits `ON_TOAST_MESSAGE`
3. Removed `pendingOrderPopup` middleware since we already have `OrdersNotificationsUpdater`
4. Got rid of `ON_POSTED_ETH_FLOW_ORDER` and added `isEthFlow` to `ON_POSTED_ORDER`
5. A bit improved `currencyAmountToTokenAmount` interface

image

  # To Test

1. Submitted order notification should be displayed as before for any kind of trade
2. `ON_TOAST_MESSAGE` event must be fired along with displaying the notification

See https://github.com/cowprotocol/cowswap/pull/3934

### CODE REVIEW

File: libs/events/src/types/orders.ts [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+  symbol: string
+  logoURI?: string
+}
+
+export type OnPostedOrderPayload = {
+  orderUid: string
+  chainId: SupportedChainId
+  owner: string
+  kind: OrderKind
+  // TODO: use UiOrderType
```
Comment (shoom3301) [inline libs/events/src/types/orders.ts:23]: Will be possible after https://github.com/cowprotocol/cowswap/pull/3901

**Comment by alfetopito** [comment]
BTW, unit tests are broken

**Comment by elena-zh** [comment]
Hey @shoom3301 , great, but: 
1. Buy orders show an incorrect message (limit/swap): 
buy PR
should be this: 
buy order toast

2. ETH-flow: shows wrapped token instead of the native
xdai

File: apps/cowswap-frontend/src/modules/swap/services/swapFlow/index.ts [hunk 2]
```diff
@@ -112,7 +113,18 @@ export async function swapFlow(
...
       : presignOrderStep(orderUid, input.contract))
 
-    EVENT_EMITTER.emit(CowEvents.ON_POSTED_ORDER, { orderUid, chainId })
+    emitPostedOrderEvent({
+      chainId,
+      id: orderUid,
+      orderCreationHash: presignTx?.hash,
+      kind: OrderKind.SELL,
+      receiver: recipientAddressOrName,
+      // TODO: check, should we use inputAmountWithSlippage instead?
```
[the remark this answers is not in the document]
Comment (anxolin) [reply apps/cowswap-frontend/src/modules/swap/services/swapFlow/index.ts:122]: Yes, I think is fine as it is. Alternativelly we could post both, and display both (showing the expected and the minum receive), but for now I think is fine

**Review by anxolin — APPROVED** [review]
Great work!

File: libs/events/src/types/orders.ts [hunk 3]
```diff
@@ -1,17 +1,36 @@
...
-  txHash: string
+export type OnPostedOrderPayload = {
+  orderUid: string
+  orderCreationHash?: string
+  chainId: SupportedChainId
+  owner: string
+  kind: OrderKind
+  orderType: UiOrderType
+  inputAmount: bigint
+  outputAmount: bigint
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply libs/events/src/types/orders.ts:27]: You are right, currently it's only used there.
But this is a payload of `@cowprotocol/events` library, I prefer using native bigint for public interfaces.

File: apps/cowswap-frontend/src/legacy/state/orders/middleware/emitPostedOrderEvent.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  kind: OrderKind
+  uiOrderType: UiOrderType
+  receiver: Nullish<string>
+  inputAmount: CurrencyAmount<Token>
+  outputAmount: CurrencyAmount<Token>
+  orderCreationHash?: string
+  isHidden?: boolean
+}
+
+export function emitPostedOrderEvent(params: PendingOrderNotificationParams) {
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/state/orders/middleware/emitPostedOrderEvent.tsx]: I've intentionally left it in `modules/orders` to show that this function is related to `orders` domain

**Pull request merged**